token-metadata-creator/test/index.spec.js

```diff
@@ -193,6 +192,7 @@ describe("token-metadata-creator", () => {
     it("Keep signatures on edit to same value", () => {
       writeTmpFile(keyFiles[alice][0], keys[alice][0]);
       cli(alice, "--name", "foo")
+      cli(alice, "--policy", policyFiles[alice]);
```

Comment:
So this test was failing, because signature is now not added on attesting when there is no policy field. Updated policy in this case. Question: is this expected?

If so then attestation step is not required when adding metadata without policy?
If so https://github.com/cardano-foundation/cardano-token-registry/wiki/How-to-prepare-an-entry-for-the-registry-%28Plutus-script%29 needs to be updated.

Reply:
First of all, sorry for overlooking adequate testing in my approval of the original PR :(

I think you're correct, the attestation step is not required when adding metadata without a policy (i.e. Plutus scripts). I think there's an issue to change that page in the wiki here: https://github.com/cardano-foundation/cardano-token-registry/issues/312. Unfortunately I can't view the contents of the changes, so I've dropped a question in that thread.

Otherwise everything you've done looks good and I've tested locally on my machine. I just have one small suggestion: https://github.com/input-output-hk/offchain-metadata-tools/pull/36

Reply:
No worries, thanks! :)

Reply:
@sevanspowell from my interpretation of the following comment:

https://github.com/input-output-hk/offchain-metadata-tools/pull/34#issuecomment-895416122

"Monetary policy can be Plutus scripts, and when they are, there's no longer such a strict relationship with key owners"
"the verification will be done client side by the end-user's wallet, using a set of trusted keys managed by the user. Wallets will be shipped with a pre-registered set of well-known keys and users will have the ability to add new ones."

So in the case of no policy, the registry will not validate the attestations, but they are still required since clients (end-user's wallet etc) will be required to validate them. - this is my understanding

Reply:
>So in the case of no policy, the registry will not validate the attestations, but they are still required since clients (end-user's wallet etc) will be required to validate them. - this is my understanding

If that is the case then current token-metadata-creator needs to be fixed I believe. In current behavior there are no signatures added on attestation when the policy is missing in the metadata file.

E.g.:

```
token-metadata-creator entry --init fda1b6b487bee2e7f64ecf24d24b1224342484c0195ee1b7b943db504d696c6c6172436f696e

token-metadata-creator entry fda1b6b487bee2e7f64ecf24d24b1224342484c0195ee1b7b943db504d696c6c6172436f696e \
  --name "PlutusCoin" \
  --description "Test Plutus token"

token-metadata-creator entry fda1b6b487bee2e7f64ecf24d24b1224342484c0195ee1b7b943db504d696c6c6172436f696e -a payment.skey

token-metadata-creator entry fda1b6b487bee2e7f64ecf24d24b1224342484c0195ee1b7b943db504d696c6c6172436f696e --finalize
```
Produces the following metadata file:
```
{
    "subject": "fda1b6b487bee2e7f64ecf24d24b1224342484c0195ee1b7b943db504d696c6c6172436f696e",
    "name": {
        "sequenceNumber": 0,
        "value": "PlutusCoin",
        "signatures": []
    },
    "policy": null,
    "description": {
        "sequenceNumber": 0,
        "value": "Test Plutus token",
        "signatures": []
    }
}
```

... and the same file will be actually created when we drop attestation step...

Also, not sure if having `policy: null` in the file is good idea. Probably the `policy` entry should just not be there...